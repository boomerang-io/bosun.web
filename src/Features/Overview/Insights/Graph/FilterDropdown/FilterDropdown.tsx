import React from "react";
import { Dropdown } from "@boomerang-io/carbon-addons-boomerang-react";

type Props = {
  label: string;
  options: any[];
  onChange: (...args: any[]) => void;
  selectedItem?: any;
};

export function FilterDropdown({ selectedItem, onChange, options, label }: Props) {
  return (
    <div style={{ width: "15rem", height: "3rem" }}>
      <Dropdown
        id="filter-dropdown"
        label={label}
        onChange={onChange}
        selectedItem={selectedItem}
        items={options}
        titleText={label}
      />
    </div>
  );
}

export default FilterDropdown;
