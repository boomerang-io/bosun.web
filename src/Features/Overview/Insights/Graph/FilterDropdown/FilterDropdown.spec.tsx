import React from "react";
import FilterDropdown from ".";
const mockfn = jest.fn();
const label = "testLabel";
const options = [{ text: "option1", value: "option1" }];
const onChange = mockfn;
describe("FilterDropdown --- Snapshot", () => {
  it("Capturing Snapshot of FilterDropdown", async () => {
    const { baseElement } = (global as any).rtlRender(
      <FilterDropdown label={label} options={options} onChange={onChange} />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
