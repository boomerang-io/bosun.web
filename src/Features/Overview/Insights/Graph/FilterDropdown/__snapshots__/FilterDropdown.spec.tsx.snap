// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDropdown --- Snapshot Capturing Snapshot of FilterDropdown 1`] = `
<body
  id="app"
>
  <div>
    <div
      style="width: 15rem; height: 3rem;"
    >
      <div
        class="bx--dropdown__wrapper bx--list-box__wrapper"
      >
        <label
          class="bx--label"
          for="downshift-0-toggle-button"
          id="downshift-0-label"
        >
          testLabel
        </label>
        <div
          class="bx--dropdown bx--list-box"
          id="filter-dropdown"
        >
          <button
            aria-disabled="false"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="downshift-0-label downshift-0-toggle-button"
            class="bx--list-box__field"
            id="downshift-0-toggle-button"
            type="button"
          >
            <span
              class="bx--list-box__label"
            >
              testLabel
            </span>
            <div
              class="bx--list-box__menu-icon"
            >
              <svg
                aria-label="Open menu"
                fill="currentColor"
                focusable="false"
                height="16"
                name="chevron--down"
                preserveAspectRatio="xMidYMid meet"
                role="img"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M8 11L3 6 3.7 5.3 8 9.6 12.3 5.3 13 6z"
                />
                <title>
                  Open menu
                </title>
              </svg>
            </div>
          </button>
          <div
            aria-labelledby="downshift-0-label"
            class="bx--list-box__menu"
            id="downshift-0-menu"
            role="listbox"
            tabindex="-1"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
